Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -349,7 +349,29 @@ extension ChannelDTO {
         // Fetch results controller requires at least one sorting descriptor.
         let sortDescriptors = query.sort.compactMap { $0.key.sortDescriptor(isAscending: $0.isAscending) }
         request.sortDescriptors = sortDescriptors.isEmpty ? [ChannelListSortingKey.defaultSortDescriptor] : sortDescriptors
-
+        request.predicate = NSCompoundPredicate(type: .and, subpredicates: channelListPredicates(query: query, chatClientConfig: chatClientConfig))
+        request.fetchLimit = query.pagination.pageSize
+        request.fetchBatchSize = query.pagination.pageSize
+        return request
+    }
+    
+    static func channelListContainsChannelFetchRequest(
+        cid: ChannelId,
+        query: ChannelListQuery,
+        chatClientConfig: ChatClientConfig
+    ) -> NSFetchRequest<ChannelDTO> {
+        let request = channelListFetchRequest(query: query, chatClientConfig: chatClientConfig)
+        let channelPredicate = NSPredicate(format: "cid == %@", cid.rawValue)
+        let channelListPredicates = channelListPredicates(query: query, chatClientConfig: chatClientConfig)
```

Comment:
Diff is slightly hard to read but I extracted existing predicates into a separate function `channelListPredicate` which is used in two places. Here I add the cid equals check.